apps/tasks/views.py

```diff
@@ -1,36 +1,73 @@
-from rest_framework import status, filters
+import http
+
+from django.conf import settings
+
+from rest_framework import filters
 from rest_framework.decorators import action
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.viewsets import ModelViewSet
 
-from apps.tasks.models import Task
-from apps.tasks.serializers import TaskSerializer, TaskAssignSerializer, DisplayTaskSerializer
+from apps.common.helpers import send_user_email
+from apps.tasks.models import Task, Comment
+from apps.tasks.serializers import TaskSerializer, TaskAssignSerializer, TaskCommentSerializer
 
 
 class TaskViewSet(ModelViewSet):
     serializer_class = TaskSerializer
     queryset = Task.objects.all()
     permission_classes = (IsAuthenticated,)
     filter_backends = [filters.SearchFilter]
-    search_fields = ['=id']
+    search_fields = ['$title']
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+    @action(detail=False, methods=['PUT'], serializer_class=TaskAssignSerializer)
+    def assign(self, request):
+        Task.objects.filter(pk=self.request.data['id']).update(assigned_to=self.request.data['assigned_to'])
+        task = Task.objects.get(pk=self.request.data['id'])
```

Comment:
Aici mai degraba poti sa folosesti metoda `self.get_object()`, ea face fix acelasi lucru, in plus arunca 404 daca nu gaseste taskul. Poti sa muti asta inainte de update ca sa validezi daca id-ul trimis se gaseste in baza si sa nu faci update degeaba daca s-a trimis un ID inexistent.

Reply:
Done